cumulusci/cli/tests/test_cci.py

```diff
@@ -2251,10 +2253,54 @@ def test_lines_from_traceback(self):
         output = cci.lines_from_traceback(content, 10)
         assert output == traceback
 
+    @mock.patch(
+        "cumulusci.cli.runtime.CliRuntime.get_org",
+        lambda *args, **kwargs: (MagicMock(), MagicMock()),
+    )
+    @mock.patch("cumulusci.core.runtime.BaseCumulusCI._load_keychain", MagicMock())
+    @mock.patch("pdb.post_mortem", MagicMock())
+    @mock.patch("cumulusci.cli.cci.tee_stdout_stderr", MagicMock())
+    @mock.patch("cumulusci.cli.cci.init_logger", MagicMock())
+    @mock.patch("cumulusci.cli.cci.get_tempfile_logger")
+    def test_run_task_debug(self, get_tempfile_logger):
+        get_tempfile_logger.return_value = (mock.Mock(), "tempfile.log")
+
+        gipnew = "cumulusci.tasks.preflight.packages.GetInstalledPackages._run_task"
+        with mock.patch(gipnew, mock_validate_debug(False)):
+            cci.main(["cci", "task", "run", "get_installed_packages"])
+        with mock.patch(gipnew, mock_validate_debug(True)):
+            cci.main(["cci", "task", "run", "get_installed_packages", "--debug"])
+
+    @mock.patch(
+        "cumulusci.cli.runtime.CliRuntime.get_org",
+        lambda *args, **kwargs: (MagicMock(), MagicMock()),
+    )
+    @mock.patch("cumulusci.core.runtime.BaseCumulusCI._load_keychain", MagicMock())
+    @mock.patch("pdb.post_mortem", MagicMock())
+    @mock.patch("cumulusci.cli.cci.tee_stdout_stderr", MagicMock())
+    @mock.patch("cumulusci.cli.cci.init_logger", MagicMock())
+    @mock.patch("cumulusci.tasks.robotframework.RobotLibDoc", MagicMock())
+    @mock.patch("cumulusci.cli.cci.get_tempfile_logger")
+    def test_run_flow_debug(self, get_tempfile_logger):
+        get_tempfile_logger.return_value = (mock.Mock(), "tempfile.log")
+        rtd = "cumulusci.tasks.robotframework.RobotTestDoc._run_task"
+
+        with mock.patch(rtd, mock_validate_debug(False)):
+            cci.main(["cci", "flow", "run", "robot_docs"])
+        with mock.patch(rtd, mock_validate_debug(True)):
+            cci.main(["cci", "flow", "run", "robot_docs", "--debug"])
```

Comment:
It's not sustainable to keep adding new tests that need to mock this many things. Doesn't need to be addressed as part of this PR, but let's try to figure out a better test harness for cci.main

Reply:
I agree and started a cleanup on the weekend.

https://github.com/SFDO-Tooling/CumulusCI/pull/2519